test/functional-api/communications/communities-discussions/discussions.it-spec.ts

```diff
@@ -0,0 +1,562 @@
+      );
+      await removeMessageFromDiscussion(
+        entitiesId.discussionId,
+        secondmessageId
+      );
+    });
+
+    test('Delete message from discussion', async () => {
+      // Arrange
+      const res = await postDiscussionComment(
+        entitiesId.discussionId,
+        'test message remove'
+      );
+
+      entitiesId.messageId = res.body.data.sendMessageToDiscussion.id;
+
+      let discussionRes = await getSpaceData(
+        entitiesId.spaceId,
+        TestUser.GLOBAL_ADMIN
+      );
+      const messagesBefore =
+        discussionRes.body.data.space.community.communication.discussions[0]
+          .messages;
+
+      // Act
+      await removeMessageFromDiscussion(
+        entitiesId.discussionId,
+        entitiesId.messageId
+      );
+
+      discussionRes = await getSpaceData(
+        entitiesId.spaceId,
+        TestUser.GLOBAL_ADMIN
+      );
+      const messagesAfter =
+        discussionRes.body.data.space.community.communication.discussions[0]
+          .messages;
+
+      // Assert
+      expect(res.statusCode).toEqual(200);
+      expect(messagesBefore).toHaveLength(1);
+      expect(messagesAfter).toHaveLength(0);
+    });
+  });
+
+  describe('Discussion messages', () => {
+    beforeAll(async () => {
+      await changePreferenceSpace(
+        entitiesId.spaceId,
+        SpacePreferenceType.ANONYMOUS_READ_ACCESS,
+        'false'
+      );
+
+      await mutation(
+        assignUserAsCommunityMember,
+        assignUserAsCommunityMemberVariablesData(
+          entitiesId.spaceCommunityId,
+          users.spaceMemberId
+        )
+      );
+
+      const discussionRes = await createDiscussion(
+        entitiesId.spaceCommunicationId,
+        'test',
+        DiscussionCategory.GENERAL
+      );
+
+      entitiesId.discussionId = discussionRes.body.data.createDiscussion.id;
+    });
+
+    afterEach(async () => {
+      await removeMessageFromDiscussion(
+        entitiesId.discussionId,
+        entitiesId.messageId
+      );
+    });
+    afterAll(async () => {
+      await deleteDiscussion(entitiesId.discussionId);
+    });
+    describe('Private Space', () => {
+      test('discussion message - PRIVATE space - read access - sender / reader (member) / reader (not member)', async () => {
+        // Arrange
+        const messageRes = await postDiscussionComment(
+          entitiesId.discussionId,
+          'PRIVATE space - admin'
+        );
+
+        entitiesId.messageId = messageRes.body.data.sendMessageToDiscussion.id;
+
+        // Act
+        const spaceDataSender = await getSpaceData(
+          entitiesId.spaceId,
+          TestUser.GLOBAL_ADMIN
+        );
+        const getMessageSender =
+          spaceDataSender.body.data.space.community.communication.discussions[0]
+            .messages;
+
+        const spaceDataReaderMember = await getSpaceData(
+          entitiesId.spaceId,
+          TestUser.HUB_MEMBER
+        );
+        const getMessageReaderMember =
```

Comment:
Also rename to something more suitable like in this case `receivedMessage` or in this case since it is a discussion `retrievedMessage`.